src/main/java/com/zufar/onlinestore/security/signin/UserAuthenticationService.java

```diff
@@ -19,12 +20,28 @@ public class UserAuthenticationService {
     private final UserDetailsService userDetailsService;
     private final JwtTokenProvider jwtTokenProvider;
     private final AuthenticationManager authenticationManager;
+    private final LoginAttemptManager loginAttemptManager;
 
     public UserAuthenticationResponse authenticate(final UserAuthenticationRequest request) {
-        authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(request.username(), request.password())
-        );
-        UserDetails user = userDetailsService.loadUserByUsername(request.username());
+        String email = request.email();
+        String password = request.password();
+
+        loginAttemptManager.validateUserLoginLockout(email);
+
+        try {
+            UsernamePasswordAuthenticationToken authenticationToken = new UsernamePasswordAuthenticationToken(email, password);
+
+            authenticationManager.authenticate(authenticationToken);
+        } catch (Exception exception) {
+            // Authentication failed, increment login attempts and lock the account if necessary
```

Comment:
i dont think we should add any comments here

Reply:
Fixed